From b05974ad28404db1c677cf5034c2ae6b06f5baf9 Mon Sep 17 00:00:00 2001
From: Lubomir Rintel <lkundrak@v3.sk>
Date: Wed, 15 Jul 2015 15:42:15 +0200
Subject: [PATCH] libnm-gtk: add symbol versions

We've added new ABI. Nevertheless, we still want to leave the rest unversioned,
so that linking against it won't add a versioned symbol dependency.

(cherry picked from commit 7ae3f7abd239beac314c75cbd5f2ee8153df0929)
---
 src/libnm-gtk/Makefile.am   |  7 ++++---
 src/libnm-gtk/libnm-gtk.ver | 19 +++++++++++++++++++
 2 files changed, 23 insertions(+), 3 deletions(-)
 create mode 100644 src/libnm-gtk/libnm-gtk.ver

diff --git a/src/libnm-gtk/Makefile.am b/src/libnm-gtk/Makefile.am
index 54cb0ac..ea15ebb 100644
--- a/src/libnm-gtk/Makefile.am
+++ b/src/libnm-gtk/Makefile.am
@@ -50,9 +50,10 @@ libnm_gtk_la_LIBADD = \
 	$(top_builddir)/src/marshallers/libmarshallers.la \
 	$(top_builddir)/src/wireless-security/libwireless-security.la
 
-libnm_gtk_la_LDFLAGS = -Wl,-no-undefined \
-	-export-symbols-regex '^nma_.*'
+EXTRA_libnm_gtk_la_DEPENDENCIES = $(srcdir)/libnm-gtk.ver
 
+libnm_gtk_la_LDFLAGS = -Wl,-no-undefined \
+	-Wl,--version-script=$(srcdir)/libnm-gtk.ver
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libnm-gtk.pc
@@ -84,4 +85,4 @@ typelib_DATA = $(INTROSPECTION_GIRS:.gir=.typelib)
 CLEANFILES = $(gir_DATA) $(typelib_DATA)
 endif
 
-EXTRA_DIST = libnm-gtk.pc.in $(ui_DATA)
+EXTRA_DIST = libnm-gtk.pc.in libnm-gtk.ver $(ui_DATA)
diff --git a/src/libnm-gtk/libnm-gtk.ver b/src/libnm-gtk/libnm-gtk.ver
new file mode 100644
index 0000000..d80cdc0
--- /dev/null
+++ b/src/libnm-gtk/libnm-gtk.ver
@@ -0,0 +1,19 @@
+libnm_gtk_1_0_0 {
+local:
+	[^n]*;
+	n[^m]*;
+	nm[^a]*;
+	nma[^_]*;
+};
+
+libnm_gtk_1_0_6 {
+global:
+	nma_utils_menu_to_secret_flags;
+	nma_utils_setup_password_storage;
+	nma_utils_update_password_storage;
+	nma_vpn_password_dialog_focus_password_ternary;
+	nma_vpn_password_dialog_get_password_ternary;
+	nma_vpn_password_dialog_set_password_ternary;
+	nma_vpn_password_dialog_set_password_ternary_label;
+	nma_vpn_password_dialog_set_show_password_ternary;
+} libnm_gtk_1_0_0;
-- 
2.4.3

